### DISCUSSION TITLE
Nested NavigationMenu / NavigationViewport

### DISCUSSION THREAD

**Discussion by saulandreee** [shadcn-ui/ui#2862] [Q&A]
Hi. I'm new to Radix UI and Shadcn UI. I'm having a problem with styling the NavigationViewport (I think :sweat_smile:  ). 

You can see my code here: https://codesandbox.io/p/github/saulandreee/backup-next-lesoin/main.
The demo here: https://fzkt7g-3300.csb.app/

Here is a short video of my problem:
[Screencast from 02-27-2024 03:25:22 PM.webm](https://github.com/shadcn-ui/ui/assets/66207635/c91959a0-b161-413e-a62a-5e59ebdc342b)

There are 2 problems. 
1. For the main navigation, I used z-[10], and z-[20] for the nested one. Then I gave z-[30] for the nested NavigationViewport, but the nested one still did not pick up the z index. 

2. The nested viewport seems to have a default height that will be used when the viewport is open, after view milliseconds, the viewport will fit the height of my content. But on that first milliseconds, the default height will affect the max height of the page, so the scrollbar will take space for a few milliseconds. How can I remove this default height?

**Comment by ismarthawkback** [comment]
hi @saulandreee Please go through this, discussion :  https://github.com/shadcn-ui/ui/discussions/4395
